### PR TITLE
chore: Updates search_deployment resource with new SDK

### PR DESCRIPTION
**Pull request by lantoli**
## Description

Updates search_deployment resource with new SDK.

- Testing improved, creation of migration test and config for it
- Removed helper method for new SDK migration: NonEmptyToPtr

Link to any related issue(s): https://jira.mongodb.org/browse/CLOUDP-222717

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contribution guidelines](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/CONTRIBUTING.md)
- [x] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code
- [x] If changes include deprecations or removals, I defined an isolated PR with a relevant title as it will be used in the auto-generated changelog.

### CODE REVIEW

File: internal/common/conversion/type_conversion.go [hunk 1]
```diff
@@ -56,11 +56,3 @@ func IsStringPresent(strPtr *string) bool {
 func MongoDBRegionToAWSRegion(region string) string {
 	return strings.ReplaceAll(strings.ToLower(region), "_", "-")
 }
-
-// NonEmptyToPtr accepts an array or slice and returns a pointer to it, except if it's empty, in that case it returns nil.
-func NonEmptyToPtr[T any](v []T) *[]T {
```
Comment (lantoli) [inline internal/common/conversion/type_conversion.go:61]: not used any more

File: internal/service/project/model_project_test.go [hunk 2]
```diff
@@ -251,7 +251,7 @@ func TestProjectDataSourceSDKToResourceTFModel(t *testing.T) {
 			project: &projectSDK,
 			projectProps: project.AdditionalProperties{
 				Teams: &admin.PaginatedTeamRole{
-					Results:    conversion.NonEmptyToPtr(teamRolesSDK),
+					Results:    &teamRolesSDK,
```
Comment (lantoli) [inline internal/service/project/model_project_test.go:254]: leftover

File: internal/service/alertconfiguration/resource_alert_configuration_test.go [hunk 3]
```diff
@@ -237,7 +237,7 @@ func TestAccConfigRSAlertConfiguration_withThresholdUpdated(t *testing.T) {
 				ImportStateIdFunc:       importStateProjectIDFunc(resourceName),
 				ImportState:             true,
 				ImportStateVerify:       true,
-				ImportStateVerifyIgnore: []string{"project_id", "matcher.0.field_name"},
+				ImportStateVerifyIgnore: []string{"project_id", "updated", "matcher.0.field_name"},
```
Comment (lantoli) [inline internal/service/alertconfiguration/resource_alert_configuration_test.go:240]: add "updated" to fix flaky test as sometimes updated time differs in a second

File: internal/service/searchdeployment/resource_search_deployment_migration_test.go [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-testing/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/hashicorp/terraform-plugin-testing/plancheck"
+	"github.com/mongodb/terraform-provider-mongodbatlas/internal/testutil/acc"
+	"github.com/mongodb/terraform-provider-mongodbatlas/internal/testutil/mig"
+)
+
+func TestAccMigrationSearchDeployment_basic(t *testing.T) {
```
Comment (AgustinBettati) [inline internal/service/searchdeployment/resource_search_deployment_migration_test.go:14]: nice!

**Review by oarbusi — APPROVED** [review]
LGTM

File: internal/service/searchdeployment/schema_search_deployment.go
Comment (AgustinBettati) [inline internal/service/searchdeployment/schema_search_deployment.go:1]: I would keep into account that we also have the file `data_source_search_deployment_schema.go`, so changing the resource schema file to `schema_search_deployment.go` can cause confusion.

Comment (lantoli) [reply internal/service/searchdeployment/schema_search_deployment.go:1]: oh, i missed that, ok I'll take it back

Comment (lantoli) [reply internal/service/searchdeployment/schema_search_deployment.go:1]: i missed that, is it ok to change it to schema_data_source_search_deployment.go ?

Comment (lantoli) [reply internal/service/searchdeployment/schema_search_deployment.go:1]: i missed that, ok I'll revert the rename

**Pull request merged**